oval_graph/_xml_parser_test_info.py

```diff
@@ -107,18 +107,37 @@ def _xml_element_to_dict(self, object_, collected_object):
         else:
             result[object_.attrib.get('id')] = "does not exist"
             result[self._get_unique_id_in_dict(
-                object_, result)] = self._get_object_items(object_)
+                object_, result)] = self._get_object_items(object_, collected_object)
         return result
 
-    def _get_object_items(self, object_):
+    def _get_object_items(self, object_, collected_object):
         out = {}
         for element in object_.iterchildren():
             if element.text and element.text.strip():
                 out[self._get_unique_id_in_dict(element, out)] = element.text
             else:
-                out[self._get_unique_id_in_dict(element, out)] = "no value"
+                if self._collected_object_is_not_none_and_not_empty_and_element_contain_var_ref(
+                        element, collected_object):
+                    variable_value = ''
+                    var_id = element.attrib.get('var_ref')
+                    for item in collected_object:
+                        if var_id == item.attrib.get('variable_id'):
+                            variable_value += item.text
+                        elif self._get_key_for_element(item) == 'message':
+                            variable_value = item.text[:item.text.find(
+                                '(') + 1] + var_id + ')'
+                    out[self._get_unique_id_in_dict(
+                        element, out)] = variable_value
+                else:
+                    out[self._get_unique_id_in_dict(element, out)] = 'no value'
         return out
 
+    def _collected_object_is_not_none_and_not_empty_and_element_contain_var_ref(
+            self, element, collected_object):
+        if collected_object is not None and 'var_ref' in element.attrib:
+            return len(collected_object)
+        return False
+
```

Comment:
The name of this function feels too long. Also, I think it's mixing 2 potentionally unrelated things. Personally I would not create a function for this. Or at least I would create a better name.